cmd/virt-launcher/virt-launcher.go

```diff
@@ -503,7 +503,12 @@ func main() {
 
 	domain := waitForDomainUUID(*qemuTimeout, events, signalStopChan, domainManager)
 	if domain != nil {
-		mon := virtlauncher.NewProcessMonitor(domain.Spec.UUID,
+		// The first argument to NewProcessMonitor will end up being grepped in /proc/*/cmdline
+		// to find what is hopefully the qemu process.
+		// If any other process includes that string, it might wrongly be considered.
+		// FIXME: maybe also look at /proc/$PID/exe to ensure it points to qemu
```

Comment:
This is what libvirt currently does when looking for the swtpm process: it calls `readlink` on `/proc/$PID/exe` and that requires `CAP_SYS_PTRACE`. If we want to drop this cap in the future an alternative approach is needed. Probably a better way would be to just look in the pid-file which libvirt creates when qemu is launched (if I recall correctly that was proposed by @rmohr during one of the weekly calls).

Reply:
Oh interesting, I didn't realize that's what they were doing!
Comment fixed, thanks!